src/main.rs

```diff
@@ -387,12 +404,74 @@ fn main() {
     info!("Selecting the Chawathe et al. (1996) edit script generator.");
 
     // Edit script generator configuration object.
-    let generator_config: Box<edit_script::EditScriptGenerator<String>> = generator_script_config(args.flag_edit);
+
+    let mut script_generator_98 = false;
+    match args.flag_edit {
+        Some(EditScriptGenerator::Chawathe96) | None => {
+            script_generator_98 = false;
+        }
+        Some(EditScriptGenerator::Chawathe98) => {
+            fn input_from_user() -> Vec<usize> {
+                println!("If the wrong values are entered, default values would be taken");
+                print!("\nPlease enter cost for edges: Insert, Delete, Update, Move, Copy, Glue: ");
+                let mut user_input = String::new();
+                let _ = stdout().flush();
+                std::io::stdin().read_line(&mut user_input)
+                                .expect("Did not enter a correct string");
+                if let Some('\n') = user_input.chars().next_back() {
+                    user_input.pop();
+                }
+                if let Some('\r') = user_input.chars().next_back() {
+                    user_input.pop();
+                }
+                // Get all the integer input from the user. Split by space.
+                let mut vec_cost: Vec<usize> =
+                    user_input.split(" ")
+                              .filter_map(|x| x.parse::<usize>().ok())
+                              .collect();
+                // Keep only values which are greater than zero
+                println!("What user has entered => {:?}", vec_cost);
+                vec_cost
+            }
+
+            let vec_cost = input_from_user();
+            // When user inputs right values
+            if vec_cost.len() == 6 {
+                // Right values
+                chawathe98_matcher::test_chawathe_matching_actual(ast_base_clone,
+                                                                  ast_diff_clone,
+                                                                  &args.arg_base_file,
+                                                                  &args.arg_diff_file,
+                                                                  vec_cost[0],
```

Comment:
Maybe it would be neater to put the costs in their own struct, and just pass that struct to `test_chawathe_matching_actual`? Also `test_` suggest that this function is a unit test, but presumably it's not?